core/sdk-core/src/main/java/software/amazon/awssdk/core/interceptor/ExecutionAttribute.java

```diff
@@ -87,6 +87,17 @@ private ExecutionAttribute(String name, ValueStorage<T> storage) {
     public static <T, U> DerivedAttributeBuilder<T, U> derivedBuilder(String name,
                                                                       @SuppressWarnings("unused") Class<T> attributeType,
                                                                       ExecutionAttribute<U> realAttribute) {
+        return new DerivedAttributeBuilder<>(name, () -> realAttribute);
+    }
+
+    /**
+     * This is the same as {@link #derivedBuilder(String, Class, ExecutionAttribute)}, but the real attribute is loaded
+     * lazily at runtime. This is useful when the real attribute is in the same class hierarchy, to avoid initialization
+     * order problems.
+     */
+    public static <T, U> DerivedAttributeBuilder<T, U> derivedBuilder(String name,
+                                                                      @SuppressWarnings("unused") Class<T> attributeType,
+                                                                      Supplier<ExecutionAttribute<U>> realAttribute) {
```

Comment:
Spotbugs complains about the dependency. I didn't test what happens if we ignore spotbugs. I suspect it could be a weird "no class def found" because of the cyclic static initialization dependency?